optimizely/src/event_api/batched_event_dispatcher.rs

```diff
@@ -60,7 +60,11 @@ impl Default for BatchedEventDispatcher {
                 }
 
                 // Send payload if reached the batch threshold
-                if let Some(payload) = payload_option.take_if(|payload| payload.size() >= DEFAULT_BATCH_THRESHOLD) {
+                if payload_option
```

Comment:
With this change, there was also this that I had first but went with the reduced version. I am not sure why I had a let around the payload as I have changed my example I was running based on this quite a few times.
```rs
if payload_option
    .as_ref()
    .map_or(false, |payload| payload.size() >= DEFAULT_BATCH_THRESHOLD)
{
    let payload = payload_option.take();
    log::debug!("Reached DEFAULT_BATCH_THRESHOLD");
    payload.send();
    if let Some(payload) = payload {
        payload.send();
    }
}
```